llvm/tools/llvm-exegesis/lib/Target.h

```diff
@@ -170,6 +170,12 @@ class ExegesisTarget {
         "configurePerfCounter is not implemented on the current architecture");
   }
 
+  // Returns the stack register for the platform.
+  virtual unsigned getStackRegister() const {
+    report_fatal_error(
+        "getStackRegister is not implemented on the current architecture");
+  }
```

Comment:
We can probably avoid introducing a new virtual call by using `TargetLowering` in `LLVMState`.

The stack pointer is `state.getSubtargetInfo().getTargetLowering()->getStackPointerRegisterToSaveRestore();`

This would avoid having to reimplement that in `ExegesisTarget` for every architecture.